File: chess-game/chess-game.v3/web/client/chess-app/src/app/pages/game/play/play/models/pieces/piece_factory.ts
import { Side } from "../side";
import { Bishop } from "./bishop";
import { King } from "./king";
import { Knight } from "./knight";
import { Pawn } from "./pawn";
import { Queen } from "./queen";
import { Rook } from "./rook";

export class PieceFactory {
    
    static getPiece(piece_id: string, type: string) {
        switch (type.toLowerCase()) {
            case 'wp':
                return new Pawn(piece_id, Side.white);
            case 'bp':
                return new Pawn(piece_id, Side.black);
            case 'wr':
                return new Rook(piece_id, Side.white);
            case 'br':
                return new Rook(piece_id, Side.black);
            case 'wb':
                return new Bishop(piece_id, Side.white);
            case 'bb':
                return new Bishop(piece_id, Side.black);
            case 'wn':
                return new Knight(piece_id, Side.white);
            case 'bn':
                return new Knight(piece_id, Side.black);
            case 'wq':
                return new Queen(piece_id, Side.white);
            case 'bq':
                return new Queen(piece_id, Side.black);
            case 'wk':
                return new King(piece_id, Side.white);
            case 'bk':
                return new King(piece_id, Side.black);
            default:
                throw Error('Un-recognized piece type');
        }
    }
}

File: chess-game/chess-game.v3/chessapp/domain/pieces/piece_factory.py
from chessapp.domain.pieces.bishop import Bishop
from chessapp.domain.pieces.king import King
from chessapp.domain.pieces.knight import Knight
from chessapp.domain.pieces.pawn import Pawn
from chessapp.domain.pieces.piece_type import PieceType
from chessapp.domain.pieces.queen import Queen
from chessapp.domain.pieces.rook import Rook
from chessapp.domain.value_objects.piece_id import PieceId
from chessapp.domain.value_objects.side import Side


class PieceFactory(object):

    @classmethod
    def create(cls, piece_id: PieceId, side: Side, type_: PieceType):
        match type_:
            case PieceType.Pawn:
                return Pawn(piece_id, side)
            case PieceType.King:
                return King(piece_id, side)
            case PieceType.Rook:
                return Rook(piece_id, side)
            case PieceType.Queen:
                return Queen(piece_id, side)
            case PieceType.Bishop:
                return Bishop(piece_id, side)
            case PieceType.Knight:
                return Knight(piece_id, side)
        return None